src/url-params.ts

```diff
@@ -175,11 +175,24 @@ export class AnalyticsUrlParams {
 	/**
 	 * @return full query parameter string that can be appended to URLs
 	 */
-	getQueryString(destinationUrl?: URL, currentUrl?: URL): string {
+	getQueryString(destinationUrl?: URL | string, currentUrl?: URL): string {
+		// we first check if destionationUrl is a relative URL string. If it is, we exit.
+		const relativeRegex = /^(?!www\.|(?:http|ftp)s?:\/\/|[A-Za-z]:\\|\/\/).*/g;
+		if (
+			typeof destinationUrl === 'string' &&
+			destinationUrl.match(relativeRegex)
+		) {
+			return '';
+		}
+
+		// If its not a relative URL string, we make sure if its a string or an URL. It its a string, we convert it to an URL object.
+		if (typeof destinationUrl === 'string') {
+			destinationUrl = new URL(destinationUrl);
```

Comment:
We need to handle the errors here when the incoming string cannot be translated to a URL. I'm not sure if `relativeRegex` ensures that the string with no match is actually a URL.

Reply:
@gelbal Ohh yeap you are right. let me look into this.